server/games/coop.py

```diff
@@ -31,6 +33,35 @@ async def validate_game_mode_settings(self):
             "TeamSpawn": ("fixed", ValidityState.SPAWN_NOT_FIXED),
             "RevealedCivilians": ("No", ValidityState.CIVILIANS_REVEALED),
             "Difficulty": (3, ValidityState.WRONG_DIFFICULTY),
-            "Expansion": (1, ValidityState.EXPANSION_DISABLED),
+            "Expansion": ("true", ValidityState.EXPANSION_DISABLED),
         }
         await self._validate_game_options(valid_options)
+
+    async def on_game_end(self):
+        """
+        When a coop game ends, we don't expect there to be any game results.
+
+        TODO: Refactor this duplication
```

Comment:
I'm guessing the offending lines were `if not self._results: await self.mark_invalid(ValidityState.UNKNOWN_RESULT)` in `Game.on_game_end`?
Maybe that check can be moved into `Game.resolve_game_results` (which can already modify the validity state) and then you override `resolve_game_results` as no-op instead?

Maybe everything in `Game.on_game_end` pertaining to results should be moved into its own method and overridden here...